### PR TITLE
MM-15186: Migrate "WebHook.GetOutgoingByTeam" to Sync by defa

### LINKED ISSUE #10651: [Help Wanted] [MM-15186] Migrate "WebHook.GetOutgoingByTeam" to Sync by default
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

----

**Notes**: [Jira ticket](https://mattermost.atlassian.net/browse/MM-15186)

Mattermost is migrating its Store layer to be sync by default, and only use Async when needed and we're looking for contributors to help with that effort. This Help Wanted issue is to migrate the `GetOutgoingByTeam` in the `WebHook` store.

The expected way to implement it is, go to the `WebHook` store implementation in the `store/sqlstore/` directory, modify the method `GetOutgoingByTeam` to return directly an object from the `model` module, and a `*model.AppError` (removing the `store.Do` wrapper). After that, you must modify the interface defined in `store/store.go` to match with the changes. Then, you should execute `make store-mocks` to rebuild the mocks with the new interface. And finally, modify the rest of the code (tests included) to use the new interface of the function properly.

Example: mattermost/mattermost-server#10613

### PR DESCRIPTION
**Pull request by therealpuneeth20**
Summary
This PR migrates the "WebHook.GetOutgoingByTeam" to Sync by default.

Ticket Link
Fixes #10651

### CODE REVIEW

File: cmd/mattermost/commands/webhook.go [hunk 1]
```diff
@@ -102,33 +101,26 @@ func listWebhookCmdF(command *cobra.Command, args []string) error {
 		}
 
 		// Fetch all hooks with a very large limit so we get them all.
-		incomingResult := make(chan store.StoreResult, 1)
```
Comment (jespino) [inline cmd/mattermost/commands/webhook.go]: You are removing here the explicit async approach, we want to request this in an async way, see the example for this cases

File: app/webhook.go [hunk 2]
```diff
@@ -463,12 +460,11 @@ func (a *App) UpdateOutgoingWebhook(oldHook, updatedHook *model.OutgoingWebhook)
 	}
 
 	var result store.StoreResult
```
Comment (jespino) [inline app/webhook.go]: This line is no longer needed

File: cmd/mattermost/commands/webhook.go [hunk 3]
```diff
@@ -129,6 +134,7 @@ func listWebhookCmdF(command *cobra.Command, args []string) error {
 		} else {
 			CommandPrintErrorln("Unable to list outgoing webhooks for '" + args[i] + "'")
 		}
+
```
Comment (jespino) [inline cmd/mattermost/commands/webhook.go]: nit: extra not needed empty line

File: cmd/mattermost/commands/webhook.go [hunk 4]
```diff
@@ -5,10 +5,10 @@ package commands
 
 import (
 	"fmt"
+	"github.com/mattermost/mattermost-server/store"
```
Comment (DSchalla) [inline cmd/mattermost/commands/webhook.go]: Nit: I think the import order got mixed up, I think one of the CI checks will complain about it.

**Comment by hanzei** [comment]
Thanks for the PR @therealpuneeth20 :+1:

**Pull request merged**